--- STM32F411xx.svd_	2016-12-29 15:05:56.666165918 +0200
+++ STM32F411xx.svd	2016-12-29 16:59:49.036399670 +0200
@@ -1081,7 +1081,7 @@
               <name>DBG_RTC_Stop</name>
               <description>RTC stopped when Core is
               halted</description>
-              <bitOffset>10</bitOffset>
+              <bitOffset>9</bitOffset>
               <bitWidth>1</bitWidth>
             </field>
             <field>
@@ -5437,6 +5437,16 @@
           </fields>
         </register>
       </registers>
+      <interrupt>
+        <name>OTG_FS</name>
+        <description>USB On The Go FS global interrupt</description>
+        <value>67</value>
+      </interrupt>
+      <interrupt>
+        <name>OTG_FS_WKUP</name>
+        <description>USB On-The-Go FS Wakeup through EXTI line interrupt</description>
+        <value>42</value>
+      </interrupt>
     </peripheral>
     <peripheral>
       <name>OTG_FS_HOST</name>
@@ -7857,6 +7867,11 @@
           </fields>
         </register>
       </registers>
+      <interrupt>
+        <name>PVD</name>
+        <description>PVD through EXTI line detection interrupt</description>
+        <value>1</value>
+      </interrupt>
     </peripheral>
     <peripheral>
       <name>RCC</name>
@@ -7869,14 +7884,9 @@
         <usage>registers</usage>
       </addressBlock>
       <interrupt>
-        <name>I2C1_EV</name>
-        <description>I2C1 event interrupt</description>
-        <value>31</value>
-      </interrupt>
-      <interrupt>
-        <name>I2C1_ER</name>
-        <description>I2C1 error interrupt</description>
-        <value>32</value>
+        <name>RCC</name>
+        <description>RCC global interrupt</description>
+        <value>5</value>
       </interrupt>
       <registers>
         <register>
@@ -8896,6 +8906,12 @@
               <bitOffset>18</bitOffset>
               <bitWidth>1</bitWidth>
             </field>
+            <field>
+              <name>SPI5EN</name>
+              <description>SPI5 clock enable</description>
+              <bitOffset>20</bitOffset>
+              <bitWidth>1</bitWidth>
+            </field>
           </fields>
         </register>
         <register>
@@ -9189,6 +9205,13 @@
               <bitOffset>18</bitOffset>
               <bitWidth>1</bitWidth>
             </field>
+            <field>
+              <name>SPI5LPEN</name>
+              <description>SPI5 clock enable during Sleep
+              mode</description>
+              <bitOffset>20</bitOffset>
+              <bitWidth>1</bitWidth>
+            </field>
           </fields>
         </register>
         <register>
@@ -9412,16 +9435,6 @@
         <size>0x400</size>
         <usage>registers</usage>
       </addressBlock>
-      <interrupt>
-        <name>I2C2_EV</name>
-        <description>I2C2 event interrupt</description>
-        <value>33</value>
-      </interrupt>
-      <interrupt>
-        <name>I2C2_ER</name>
-        <description>I2C2 error interrupt</description>
-        <value>34</value>
-      </interrupt>
       <registers>
         <register>
           <name>TR</name>
@@ -10744,6 +10757,16 @@
           </fields>
         </register>
       </registers>
+      <interrupt>
+        <name>RTC_Alarm</name>
+        <description>RTC Alarms (A and B) through EXTI line interrupt</description>
+        <value>41</value>
+      </interrupt>
+      <interrupt>
+        <name>RTC_WKUP</name>
+        <description>RTC Wakeup interrupt through the EXTI line</description>
+        <value>3</value>
+      </interrupt>
     </peripheral>
     <peripheral>
       <name>SDIO</name>
@@ -10757,14 +10780,9 @@
         <usage>registers</usage>
       </addressBlock>
       <interrupt>
-        <name>I2C3_EV</name>
-        <description>I2C3 event interrupt</description>
-        <value>72</value>
-      </interrupt>
-      <interrupt>
-        <name>I2C3_ER</name>
-        <description>I2C3 error interrupt</description>
-        <value>73</value>
+        <name>SDIO</name>
+        <description>SDIO global interrupt</description>
+        <value>49</value>
       </interrupt>
       <registers>
         <register>
@@ -12920,10 +12938,29 @@
           </fields>
         </register>
       </registers>
-    </peripheral>
-    <peripheral derivedFrom="TIM1">
-      <name>TIM8</name>
-      <baseAddress>0x40010400</baseAddress>
+      <interrupt>
+        <name>TIM1_BRK_TIM9</name>
+        <description>TIM1 Break interrupt and TIM9 global
+        interrupt</description>
+        <value>24</value>
+      </interrupt>
+      <interrupt>
+        <name>TIM1_UP_TIM10</name>
+        <description>TIM1 Update interrupt and TIM10 global
+        interrupt</description>
+        <value>25</value>
+      </interrupt>
+      <interrupt>
+        <name>TIM1_TRG_COM_TIM11</name>
+        <description>TIM1 Trigger and Commutation interrupts and
+        TIM11 global interrupt</description>
+        <value>26</value>
+      </interrupt>
+      <interrupt>
+        <name>TIM1_CC</name>
+        <description>TIM1 Capture Compare interrupt</description>
+        <value>27</value>
+      </interrupt>
     </peripheral>
     <peripheral>
       <name>TIM10</name>
@@ -12936,9 +12973,10 @@
         <usage>registers</usage>
       </addressBlock>
       <interrupt>
-        <name>SPI1</name>
-        <description>SPI1 global interrupt</description>
-        <value>35</value>
+        <name>TIM1_UP_TIM10</name>
+        <description>TIM1 Update interrupt and TIM10 global
+        interrupt</description>
+        <value>25</value>
       </interrupt>
       <registers>
         <register>
@@ -13246,9 +13284,10 @@
         <usage>registers</usage>
       </addressBlock>
       <interrupt>
-        <name>SPI2</name>
-        <description>SPI2 global interrupt</description>
-        <value>36</value>
+        <name>TIM1_TRG_COM_TIM11</name>
+        <description>TIM1 Trigger and Commutation interrupts and
+        TIM11 global interrupt</description>
+        <value>26</value>
       </interrupt>
       <registers>
         <register>
@@ -13574,9 +13613,9 @@
         <usage>registers</usage>
       </addressBlock>
       <interrupt>
-        <name>SPI3</name>
-        <description>SPI3 global interrupt</description>
-        <value>51</value>
+        <name>TIM2</name>
+        <description>TIM2 global interrupt</description>
+        <value>28</value>
       </interrupt>
       <registers>
         <register>
@@ -14515,9 +14554,9 @@
         <usage>registers</usage>
       </addressBlock>
       <interrupt>
-        <name>SPI4</name>
-        <description>SPI4 global interrupt</description>
-        <value>84</value>
+        <name>TIM3</name>
+        <description>TIM3 global interrupt</description>
+        <value>29</value>
       </interrupt>
       <registers>
         <register>
@@ -15431,6 +15470,11 @@
     <peripheral derivedFrom="TIM3">
       <name>TIM4</name>
       <baseAddress>0x40000800</baseAddress>
+      <interrupt>
+        <name>TIM4</name>
+        <description>TIM4 global interrupt</description>
+        <value>30</value>
+      </interrupt>
     </peripheral>
     <peripheral>
       <name>TIM5</name>
@@ -16367,6 +16411,11 @@
           </fields>
         </register>
       </registers>
+      <interrupt>
+        <name>TIM5</name>
+        <description>TIM5 global interrupt</description>
+        <value>50</value>
+      </interrupt>
     </peripheral>
     <peripheral>
       <name>TIM9</name>
@@ -16854,6 +16903,11 @@
           </fields>
         </register>
       </registers>
+      <interrupt>
+        <name>TIM1_BRK_TIM9</name>
+        <description>TIM1 Break interrupt and TIM9 global interrupt</description>
+        <value>24</value>
+      </interrupt>
     </peripheral>
     <peripheral>
       <name>USART1</name>
@@ -16867,16 +16921,10 @@
         <usage>registers</usage>
       </addressBlock>
       <interrupt>
-        <name>OTG_FS_WKUP</name>
-        <description>USB On-The-Go FS Wakeup through EXTI line
-        interrupt</description>
-        <value>42</value>
-      </interrupt>
-      <interrupt>
-        <name>OTG_FS</name>
-        <description>USB On The Go FS global
+        <name>USART1</name>
+        <description>USART1 global
         interrupt</description>
-        <value>67</value>
+        <value>37</value>
       </interrupt>
       <registers>
         <register>
@@ -17282,10 +17330,22 @@
     <peripheral derivedFrom="USART1">
       <name>USART2</name>
       <baseAddress>0x40004400</baseAddress>
+      <interrupt>
+        <name>USART2</name>
+        <description>USART2 global
+        interrupt</description>
+        <value>38</value>
+      </interrupt>
     </peripheral>
     <peripheral derivedFrom="USART1">
       <name>USART6</name>
       <baseAddress>0x40011400</baseAddress>
+      <interrupt>
+        <name>USART6</name>
+        <description>USART6 global
+        interrupt</description>
+        <value>71</value>
+      </interrupt>
     </peripheral>
     <peripheral>
       <name>WWDG</name>
@@ -17298,10 +17358,10 @@
         <usage>registers</usage>
       </addressBlock>
       <interrupt>
-        <name>PVD</name>
-        <description>PVD through EXTI line detection
+        <name>WWDG</name>
+        <description>Window Watchdog
         interrupt</description>
-        <value>1</value>
+        <value>0</value>
       </interrupt>
       <registers>
         <register>
@@ -17393,9 +17453,44 @@
         <usage>registers</usage>
       </addressBlock>
       <interrupt>
-        <name>RCC</name>
-        <description>RCC global interrupt</description>
-        <value>5</value>
+        <name>DMA2_Stream0</name>
+        <description>DMA2 Stream0 global interrupt</description>
+        <value>56</value>
+      </interrupt>
+      <interrupt>
+        <name>DMA2_Stream1</name>
+        <description>DMA2 Stream1 global interrupt</description>
+        <value>57</value>
+      </interrupt>
+      <interrupt>
+        <name>DMA2_Stream2</name>
+        <description>DMA2 Stream2 global interrupt</description>
+        <value>58</value>
+      </interrupt>
+      <interrupt>
+        <name>DMA2_Stream3</name>
+        <description>DMA2 Stream3 global interrupt</description>
+        <value>59</value>
+      </interrupt>
+      <interrupt>
+        <name>DMA2_Stream4</name>
+        <description>DMA2 Stream4 global interrupt</description>
+        <value>60</value>
+      </interrupt>
+      <interrupt>
+        <name>DMA2_Stream5</name>
+        <description>DMA2 Stream5 global interrupt</description>
+        <value>68</value>
+      </interrupt>
+      <interrupt>
+        <name>DMA2_Stream6</name>
+        <description>DMA2 Stream6 global interrupt</description>
+        <value>69</value>
+      </interrupt>
+      <interrupt>
+        <name>DMA2_Stream7</name>
+        <description>DMA2 Stream7 global interrupt</description>
+        <value>70</value>
       </interrupt>
       <registers>
         <register>
@@ -20036,16 +20131,44 @@
       <name>DMA1</name>
       <baseAddress>0x40026000</baseAddress>
       <interrupt>
-        <name>RTC_WKUP</name>
-        <description>RTC Wakeup interrupt through the EXTI
-        line</description>
-        <value>3</value>
+        <name>DMA1_Stream0</name>
+        <description>DMA1 Stream0 global interrupt</description>
+        <value>11</value>
       </interrupt>
       <interrupt>
-        <name>RTC_Alarm</name>
-        <description>RTC Alarms (A and B) through EXTI line
-        interrupt</description>
-        <value>41</value>
+        <name>DMA1_Stream1</name>
+        <description>DMA1 Stream1 global interrupt</description>
+        <value>12</value>
+      </interrupt>
+      <interrupt>
+        <name>DMA1_Stream2</name>
+        <description>DMA1 Stream2 global interrupt</description>
+        <value>13</value>
+      </interrupt>
+      <interrupt>
+        <name>DMA1_Stream3</name>
+        <description>DMA1 Stream3 global interrupt</description>
+        <value>14</value>
+      </interrupt>
+      <interrupt>
+        <name>DMA1_Stream4</name>
+        <description>DMA1 Stream4 global interrupt</description>
+        <value>15</value>
+      </interrupt>
+      <interrupt>
+        <name>DMA1_Stream5</name>
+        <description>DMA1 Stream5 global interrupt</description>
+        <value>16</value>
+      </interrupt>
+      <interrupt>
+        <name>DMA1_Stream6</name>
+        <description>DMA1 Stream6 global interrupt</description>
+        <value>17</value>
+      </interrupt>
+      <interrupt>
+        <name>DMA1_Stream7</name>
+        <description>DMA1 Stream7 global interrupt</description>
+        <value>47</value>
       </interrupt>
     </peripheral>
     <peripheral>
@@ -20058,11 +20181,6 @@
         <size>0x400</size>
         <usage>registers</usage>
       </addressBlock>
-      <interrupt>
-        <name>SDIO</name>
-        <description>SDIO global interrupt</description>
-        <value>49</value>
-      </interrupt>
       <registers>
         <register>
           <name>MODER</name>
@@ -21316,39 +21434,10 @@
     <peripheral derivedFrom="GPIOH">
       <name>GPIOD</name>
       <baseAddress>0X40020C00</baseAddress>
-      <interrupt>
-        <name>TIM1_BRK_TIM9</name>
-        <description>TIM1 Break interrupt and TIM9 global
-        interrupt</description>
-        <value>24</value>
-      </interrupt>
-      <interrupt>
-        <name>TIM1_UP_TIM10</name>
-        <description>TIM1 Update interrupt and TIM10 global
-        interrupt</description>
-        <value>25</value>
-      </interrupt>
-      <interrupt>
-        <name>TIM1_TRG_COM_TIM11</name>
-        <description>TIM1 Trigger and Commutation interrupts and
-        TIM11 global interrupt</description>
-        <value>26</value>
-      </interrupt>
-      <interrupt>
-        <name>TIM1_CC</name>
-        <description>TIM1 Capture Compare interrupt</description>
-        <value>27</value>
-      </interrupt>
     </peripheral>
     <peripheral derivedFrom="GPIOH">
       <name>GPIOC</name>
       <baseAddress>0x40020800</baseAddress>
-      <interrupt>
-        <name>TIM1_UP_TIM10</name>
-        <description>TIM1 Update interrupt and TIM10 global
-        interrupt</description>
-        <value>25</value>
-      </interrupt>
     </peripheral>
     <peripheral>
       <name>GPIOB</name>
@@ -21360,12 +21449,6 @@
         <size>0x400</size>
         <usage>registers</usage>
       </addressBlock>
-      <interrupt>
-        <name>TIM1_TRG_COM_TIM11</name>
-        <description>TIM1 Trigger and Commutation interrupts and
-        TIM11 global interrupt</description>
-        <value>26</value>
-      </interrupt>
       <registers>
         <register>
           <name>MODER</name>
@@ -22622,11 +22705,6 @@
         <size>0x400</size>
         <usage>registers</usage>
       </addressBlock>
-      <interrupt>
-        <name>TIM2</name>
-        <description>TIM2 global interrupt</description>
-        <value>28</value>
-      </interrupt>
       <registers>
         <register>
           <name>MODER</name>
@@ -23884,9 +23962,14 @@
         <usage>registers</usage>
       </addressBlock>
       <interrupt>
-        <name>TIM3</name>
-        <description>TIM3 global interrupt</description>
-        <value>29</value>
+        <name>I2C3_ER</name>
+        <description>I2C3 error interrupt</description>
+        <value>73</value>
+      </interrupt>
+      <interrupt>
+        <name>I2C3_EV</name>
+        <description>I2C3 event interrupt</description>
+        <value>72</value>
       </interrupt>
       <registers>
         <register>
@@ -24341,28 +24424,28 @@
       <name>I2C2</name>
       <baseAddress>0x40005800</baseAddress>
       <interrupt>
-        <name>I2C3_EV</name>
-        <description>I2C3 event interrupt</description>
-        <value>72</value>
+        <name>I2C2_EV</name>
+        <description>I2C2 event interrupt</description>
+        <value>33</value>
       </interrupt>
       <interrupt>
-        <name>I2C3_ER</name>
-        <description>I2C3 error interrupt</description>
-        <value>73</value>
+        <name>I2C2_ER</name>
+        <description>I2C2 error interrupt</description>
+        <value>34</value>
       </interrupt>
     </peripheral>
     <peripheral derivedFrom="I2C3">
       <name>I2C1</name>
       <baseAddress>0x40005400</baseAddress>
       <interrupt>
-        <name>I2C2_EV</name>
-        <description>I2C2 event interrupt</description>
-        <value>33</value>
+        <name>I2C1_EV</name>
+        <description>I2C1 event interrupt</description>
+        <value>31</value>
       </interrupt>
       <interrupt>
-        <name>I2C2_ER</name>
-        <description>I2C2 error interrupt</description>
-        <value>34</value>
+        <name>I2C1_ER</name>
+        <description>I2C1 error interrupt</description>
+        <value>32</value>
       </interrupt>
     </peripheral>
     <peripheral>
@@ -24375,16 +24458,6 @@
         <size>0x400</size>
         <usage>registers</usage>
       </addressBlock>
-      <interrupt>
-        <name>I2C1_EV</name>
-        <description>I2C1 event interrupt</description>
-        <value>31</value>
-      </interrupt>
-      <interrupt>
-        <name>I2C1_ER</name>
-        <description>I2C1 error interrupt</description>
-        <value>32</value>
-      </interrupt>
       <registers>
         <register>
           <name>CR1</name>
@@ -24781,10 +24854,6 @@
     <peripheral derivedFrom="I2S2ext">
       <name>SPI1</name>
       <baseAddress>0x40013000</baseAddress>
-    </peripheral>
-    <peripheral derivedFrom="I2S2ext">
-      <name>SPI2</name>
-      <baseAddress>0x40003800</baseAddress>
       <interrupt>
         <name>SPI1</name>
         <description>SPI1 global interrupt</description>
@@ -24792,8 +24861,8 @@
       </interrupt>
     </peripheral>
     <peripheral derivedFrom="I2S2ext">
-      <name>SPI3</name>
-      <baseAddress>0x40003C00</baseAddress>
+      <name>SPI2</name>
+      <baseAddress>0x40003800</baseAddress>
       <interrupt>
         <name>SPI2</name>
         <description>SPI2 global interrupt</description>
@@ -24801,8 +24870,8 @@
       </interrupt>
     </peripheral>
     <peripheral derivedFrom="I2S2ext">
-      <name>SPI4</name>
-      <baseAddress>0x40013400</baseAddress>
+      <name>SPI3</name>
+      <baseAddress>0x40003C00</baseAddress>
       <interrupt>
         <name>SPI3</name>
         <description>SPI3 global interrupt</description>
@@ -24810,8 +24879,22 @@
       </interrupt>
     </peripheral>
     <peripheral derivedFrom="I2S2ext">
+      <name>SPI4</name>
+      <baseAddress>0x40013400</baseAddress>
+      <interrupt>
+        <name>SPI4</name>
+        <description>SPI4 global interrupt</description>
+        <value>84</value>
+      </interrupt>
+    </peripheral>
+    <peripheral derivedFrom="I2S2ext">
       <name>SPI5</name>
       <baseAddress>0x40015000</baseAddress>
+      <interrupt>
+        <name>SPI5</name>
+        <description>SPI5 global interrupt</description>
+        <value>85</value>
+      </interrupt>
     </peripheral>
     <peripheral>
       <name>NVIC</name>
